tsl/src/nodes/decompress_chunk/exec.c

```diff
@@ -317,35 +322,56 @@ initialize_batch(DecompressChunkState *state, TupleTableSlot *slot)
 		{
 			case COMPRESSED_COLUMN:
 			{
-				value = slot_getattr(slot, column->compressed_scan_attno, &isnull);
-				if (!isnull)
-				{
-					CompressedDataHeader *header = (CompressedDataHeader *) PG_DETOAST_DATUM(value);
+				column->compressed.iterator = NULL;
 
-					column->compressed.iterator =
-						tsl_get_decompression_iterator_init(header->compression_algorithm,
-															state->reverse)(PointerGetDatum(header),
-																			column->typid);
-				}
-				else
+				value = slot_getattr(compressed_slot, column->compressed_scan_attno, &isnull);
+
+				if (isnull)
+				{
+					/*
+					 * The column will have a default value for the entire batch,
+					 * set it now.
+					 */
```

Comment:
Is the line below still needed? The iterator was already set to `NULL` in line 325.

Reply:
Removed.